src/picknik_ur_studio_integration_testing/picknik_ur_studio_integration_testing/generate_agent_plus_drivers_launch_description.py

```diff
@@ -0,0 +1,211 @@
+    test_objective_path = Path(
+        get_package_share_directory("moveit_studio_integration_testing"),
+        "test",
+        "objectives",
+    )
+
+    # The agent looks for objective XML files in the user folder
+    # <HOME>/.config/moveit_studio/objectives
+    # This folder is created by the REST API which is executed before the agent.
+    # While testing the agent alone, this folder does not exist so we need to
+    # point the agent to an alternate set of locations defined by test_objective_path.
+    # Additionally, we must convert this list of locations to a comma separated list
+    # because standard lists cannot be used as launch arguments.
+    objective_library_directories = ",".join(
+        system_config_parser.get_objective_library_paths() + [str(test_objective_path)]
+    )
+
+    # Get launch description for Studio Agent nodes
+    agent_launch_description = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            PathJoinSubstitution(
+                [
+                    FindPackageShare("moveit_studio_agent"),
+                    "launch",
+                    "studio_agent.launch.py",
+                ]
+            ),
+        ),
+        launch_arguments={
+            "is_test": "true",
+            "objective_library_directories": objective_library_directories,
+        }.items(),
+    )
+
+    # Get launch description for Studio robot driver nodes
+    robot_driver_launch_description = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            PathJoinSubstitution(
+                [
+                    FindPackageShare("moveit_studio_agent"),
+                    "launch",
+                    "robot_drivers.launch.py",
+                ]
+            )
+        )
+    )
+
+    # Add test executable
+    objective_client_gtest = launch_ros.actions.Node(
+        executable=PathJoinSubstitution(
+            [LaunchConfiguration("test_binary_dir"), gtest_name]
+        ),
+        output="both",
+    )
+
+    # Create the directory in which rosbag output will be saved.
+    make_rosbag_directory = ExecuteProcess(
+        cmd=[
+            "mkdir",
+            "-p",
+            TextSubstitution(text=rosbag_output_path),
+        ],
+        output="both",
+        condition=condition_save_rosbag,
+    )
+
+    # Record a rosbag file containing MoveIt planning scene updates.
+    rosbag_record = ExecuteProcess(
+        cmd=[
+            "ros2",
+            "bag",
+            "record",
+            "--output",
+            PathJoinSubstitution(
+                [
+                    TextSubstitution(text=rosbag_output_path),
+                    gtest_name + "_" + timestamp_string,
+                ]
+            ),
+            "/monitored_planning_scene",
+        ],
+        output="both",
+        condition=condition_save_rosbag,
+    )
+
+    # TODO(9427): Add support for taking snapshots prior to test execution.
```

Comment:
I think this number corresponds to Shortcut stories which we are no longer using, can probably remove